BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/BlueCollectionOnDisk.java

```diff
@@ -150,4 +151,25 @@ public void reportWrites(List<RollupTarget> rollupTargets) {
 	public RollupScheduler getRollupScheduler() {
 		return rollupScheduler;
 	}
+
+	@Override
+	public CollectionMetaData getMetaData() {
+		return metadata;
+	}
+
+	private CollectionMetaData metadata;
+
+	@Override
+	protected CollectionMetaData getOrCreateMetadata() {

```

Comment:
Why getOrCreate instead of just setting the meta data object in the constructor and returning it in the getter?

Reply:
so it needs to be created in the superclass by a method in the subclasses so that the superclass constructor can use it.  And it needs to be held in the subclass.  So we could do something like setMetadata(createMetadata()) but that seems even more awkward to me.  If you have a cleaner idea,  I'd be all for it.